src/server/api/general.js

```diff
@@ -133,6 +134,18 @@ export default function useGeneralApi(app) {
                 );
             }
 
+            // Is an account creation already happening for this user, maybe on another request?
+            // If so, throw an error.
+            if (user.creation_hash !== null) {
+                throw new Error('An account creation is in progress');
+            }
+            // If not, set this user's creation_hash.
+            // We'll check this later on, just before we create the account on chain.
+            const creationHash = hash
```

Comment:
couldn't this just be `crypto.randomBytes(32)`? the first rule of entropy exhaustion is  use as little entropy as you can get away with

Reply:
i was kinda wondering how big to make this :) i'll change it to 32